plugin-server/src/worker/ingestion/properties-updater.ts

```diff
@@ -66,60 +85,133 @@ export async function updatePersonProperties(
     }
 }
 
-export function calculateUpdatedProperties(
-    person: Person,
+export async function upsertGroup(
+    db: DB,
+    teamId: TeamId,
+    groupTypeIndex: GroupTypeIndex,
+    groupKey: string,
+    properties: Properties,
+    timestamp: DateTime
+): Promise<void> {
+    try {
+        const [propertiesUpdate, createdAt, version] = await db.postgresTransaction(async (client) => {
+            const group: Group | undefined = await db.fetchGroup(teamId, groupTypeIndex, groupKey, client, {
+                forUpdate: true,
+            })
+            const createdAt = group?.created_at || timestamp
+            const version = (group?.version || 0) + 1
+
+            const propertiesUpdate = calculateUpdate(
```

Comment:
do we need to do this if we haven't found a group in the db? we'll anyway come back here if `RaceConditionError` is thrown

Reply:
this comment is superseded by the one about this transaction as a whole